### PR TITLE
Update Loki to 2.1.2

### PR DESCRIPTION
**Pull request by ssmiller25**
Updating Loki to the latest version, and minor updates to make compatible with the updated Prometheus stack operator.

Thank you for wanting to submit a Pull Request to the Civo Kubernetes Marketplace repository! 

If your pull request concerns an existing Marketplace application, please make sure you have:

* [X] Notified the Maintainer of the application in this pull request so that they are aware of your proposal. - Yes, Myself.
* [X] Outlined the changes you are proposing, and the reasons these are required (e.g. stability, compatibility with new versions of Kubernetes implementations, etc).
* [X] Tested that the application works with the proposed updates applied (including a screenshot)

Testing Framework:
```sh
me@penguin:~/git/kubernetes-marketplace/loki$ make test-keep
Creating loki-test
The cluster loki-test (d9d833f7-0256-48b9-b5b7-622fb8e986d3) has been created in 1 min 36 sec
Waiting for all 3 nodes to come up
Test 1
Test 2
Test 3
Applying manifest and test run
namespace/loki created
helmchart.helm.cattle.io/loki created
job.batch/testjob created
Checking status of test run
Test 1
Test 2
Test 3
Test 4
Test 5
Test 6
Test 7
Success!
```

Screenshot in operational cluster:

loki-interface

### CODE REVIEW

**Comment by DoNnMyTh** [comment]
@ssmiller25 
loki/manifest.yaml has a dependency on "kube-prometheus-stack" but that application doesn't exist to fix it you will need to use "prometheus-operator" as there is no app as "kube-prometheus-stack" in marketplace

File: loki/manifest.yaml [hunk 1]
```diff
@@ -1,10 +1,10 @@
 ---
 name: loki-stack
 title: "Loki Stack "
-version: v0.41.2
+version: v2.1.2
 maintainer: "me@r15cookie.com"
 dependencies:
-  - prometheus-operator
+  - kube-prometheus-stack
```
Comment (DoNnMyTh) [inline loki/manifest.yaml:7]: loki/manifest.yaml has a dependency on "kube-prometheus-stack" but that application doesn't exist to fix this error you have to use "prometheus-operator" as there is no app named as "kube-prometheus-stack" in marketplace and dependencies look for app in marketplace

**Review by DoNnMyTh — CHANGES_REQUESTED** [review]
Please fix the error below

File: loki/manifest.yaml (hunk 1 above)
Comment (ssmiller25) [reply loki/manifest.yaml:7]: Thanks for taking a look!  This used to be set to "prometheus-operator", but my test scripts failed [had to update my test cluster to specify kube-prometheus-stack](https://github.com/ssmiller25/kubernetes-marketplace/blob/5acef324bf94ad016617cc090aec706adcdfd916/loki/Makefile#L68).  When I looked, it seems that promethues-operator isn't offered as an app, and only kube-prometheus-stack is available via the CLI.  I can move it back to prometheus-operator if that will work, but  confusing that I can't request that application to be installed by that name via the CLI. 

```sh
me@penguin:~/git/kubernetes-marketplace$ civo k3s application list | grep -i prometheus
| dynamic-pv-scaler         | 0.1.0            | storage      |                                                                                                                                              | prometheus-operator                 |
| kube-prometheus-stack     | 0.38.1           | monitoring   |                                                                                                                                              |                                     |
| loki-stack                | v0.41.2          | monitoring   |                                                                                                                                              | prometheus-operator                 |
```

**Review by DoNnMyTh — APPROVED** [review]
@saiyam1814 It can be merged now

**Comment by saiyam1814** [comment]
LGTM

**Pull request merged**